### DISCUSSION TITLE
Launching memtest86+ from systemd-boot

### DISCUSSION THREAD

**Discussion by nyddogghr** [memtest86plus/memtest86plus#414] [Q&A]
Hi,
I'm trying to launch memtest86+ from systemd-boot (EFI).
I've seen https://github.com/memtest86plus/memtest86plus/discussions/199 already and tried to do the same (boot entry is also mentionned on https://wiki.archlinux.org/title/Systemd-boot#Memtest86+). Here it is, along with my archlinux working entry.
![image](https://github.com/memtest86plus/memtest86plus/assets/3328219/f1e5c596-250c-4d82-8659-b24912b0a3d8)
I do have secure boot disabled and the following systemd boot status:
![image](https://github.com/memtest86plus/memtest86plus/assets/3328219/0b39e033-b57a-4fca-aa94-21ff9e6f1377)
However no matter how many times I tried, I only get a black screen, keyboard/mouse are powered on, nothing happens until I hit the reset button on my computer.
What am I missing?
I can indeed run memtest86+ when using a live usb of archlinux, that uses grub instead of systemd-boot.
Thanks

**Comment by martinwhitaker** [comment]
I've never used systemd-boot before, but I've just tried it in a VM using the same configuration as you, and it boots Memtest86+ without any problem.

If you can build and install a `memtest.efi` image from the latest source code in the main branch, try adding
````
options efidebug
````
to your `memtest86+.conf` file.

**Comment by debrouxl** [comment]
There's #265 about adding a section to the EFI images for Boot Loader Spec compliance and automatic import.

**Comment by nyddogghr** [comment]
> If you can build and install a `memtest.efi` image from the latest source code in the main branch, try adding
> 
> ```
> options efidebug
> ```
> 
> to your `memtest86+.conf` file.

I just did that, nothing changed. memtest.efi built from master (build64), option added, still blank screen.
![image](https://github.com/memtest86plus/memtest86plus/assets/3328219/098796ba-e5ab-4fcc-a0ce-71ed64532401)

> There's https://github.com/memtest86plus/memtest86plus/issues/265 about adding a section to the EFI images for Boot Loader Spec compliance and automatic import.

That would be nice indeed, but if I understand correctly it's not in master yet, right?

**Comment by wsanders** [comment]
Fedora does it this way for grub2: There is a memtest86+-7.00 rpm that supplies the EFI file and creates an entry in /boot/loader/entries with the following contents:
`title Memory test (memtest86+x64.efi)
version 0-7.00-memtest86+
linux /memtest86+x64.efi
grub_users $grub_users
grub_arg --unrestricted
grub_class memtest
`
The EFI file entry is relative to you EFI boot filesystem, so you then copy memtest86+x64.efi to /boot and it just works. The EFI is not signed (yet) so you will need to boot it with secure boot disabled.

**Reply by nyddogghr** [reply]
This looks like a configuration using grub somehow, not just systemd-boot, am I missing something?
I already tried copying the efi file, either taken from the AUR, or built from sources, and with secure boot disabled.

**Reply by wsanders** [reply]
memtest86+ will be loaded by grub, before the OS starts, so as far as I can guess systemd-boot isn't involved. But I don't know how Arc boots. The above description is for Fedora, memtest86+x64.efi resides in /boot along with the kernel, and is started instead of the kernel loading.
